NXOS: Support IPv6 ACLs

tomilukkarinen opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Add support for IPv6 ACLs in nxos_acl module. 
<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
nxos_acl

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.4.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, May  3 2017, 07:55:04) [GCC 4.8.5 20150623 (Red Hat 4.8.5-14)]
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```
ACL Example:

ipv6 access-list ipv6-acl
  10 permit tcp A:B::C:D/LEN any eq 22
```

sdodsley committed:
Add new purefa_pgsnap module

Supplies the ability to create and destroy FlashArray
protection groups.

tbouvet committed:
Fix credentials for Tower API V2

agaffney committed:
Only template each hostvars var on-demand (fixes #33259)

bcoca committed:
deal with no config for view

fixes #35965

dagwieers committed:
ACI: Assorted set of fixes to docs (#35780)

* Assorted set of fixes

* Cosmetic changes to lists

* Add more information related to connection throttling

* Changes to TOC

* Document return values

* More improvements

* Fix casing in title

dagwieers committed:
Provide more debug output on failures (#35961)

This PR includes:
- payload output on failure, when requested
- add additional kwargs to aci.exit_json()

We may want to enable some of this debug output by default on failure ?

gundalow committed:
Test against a single transport (#35903)

Network integration tests can use tags to run only a specific transport

ryansb committed:
Tower callback fixes (#35927)

* Fix tower_callback script return code

* Add ec2_key parameter to integration tests

dagwieers committed:
aci_interface_policy_leaf_policy_group: Fix tests (#35973)

dagwieers committed:
ACI: Fixes to recent change to parameter choices (#35968)

This PR includes:
- Fixes related to the recent merge of #31637 and #34537
- A generic fix for a reference for assignment issue
- Fixes to aci.boolean() in order to catch exception

s-hertel committed:
[cloud] Make ec2_vpc_route_table wait for the route to propagate (#35975)

* Stabilize ec2_vpc_route_table

Wait for route table to be present before attempting to use it

Sleep before getting the final state of the route table in case modifications are incomplete

* Conditionally wait if changes were made

* Simplify logic

nitzmahone committed:
initial cutover to API profiles (#35538)

* hardcoded API profiles in azure_rm_common
* changed azure_rm_securitygroup module to use api_profiles, dynamic models, kwargs on all SDK methods
* changed azure_rm_containerinstance module to use api_profiles, dynamic models, kwargs on all SDK methods
* fixed polling performance issue in azure_rm_securitygroup (default poll interval was 30s)

mattclay committed:
Run each Windows version independently in CI.

mattclay committed:
Rebalance cloud tests into 5 groups.

siwyd committed:
Add 'digital_ocean' as group (#21631)

BondAnthony committed:
Removed sort from json dump since response is a mix of strings and integers (#34217)

newswangerd committed:
removed to_subnet from list of imports on ios_l3_interface.py as it was not being used (#35969)

mattclay committed:
Disable unstable azure_rm_containerinstance test.

mattclay committed:
Add notify entries for files & dirs in BOTMETA.

Akasurde committed:
VMware: Additional check for RootSnapshot (#36001)

This fix adds an additional check if there is no snapshot available
in VM. `rootSnapshotList` is None, if no snapshot is taken.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

jamesv1994 committed:
Fixed spelling error in runit module (#35948)

mattclay committed:
Move requirements into tests. (#35885)

tedder committed:
fix spelling. Seriously, that's it.

mattclay committed:
Disable unstable Azure integration tests.

rvgate committed:
Change wording of Going Further section (#35949)

* Change wording of Going Further section

The original wording was confusing with its non-devel branch part.
Made it much clearer for people to understand what to do.

Also removed the git command for creating branches, I believe a
develop should be aware of how basic git commands work.

* Edited for concision and clarity.

* Typo fix

* Typo fix

Qalthos committed:
Fix VyOS check mode (#35977)

* Revert "commented out check tests (#35788)"

This reverts commit 3df2561405aae65b6592b2dc25a8b72b1804f0c7.

* `discard_changes()` now uses `exit discard`

instead of just `discard`. Also a slight tweak to avoid a second `exit` after
`exit discard`

Akasurde committed:
VMware: Minor typo in vmware_host_dns_facts (#36018)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mikewiebe committed:
Fix nxos_igmp_interface for diff nxos versions (#35959)

ganeshrn committed:
Fix vrf parsing in eos_vrf and eos_eapi module (#35791)

* Fix vrf parsing in eos_vrf and eos_eapi module

Fixes #30250

Fix logic to parse vrf when interface value in `show vrf`
command output spans on multiple lines

* Add idempotent test case

dagwieers committed:
Improve convert_bool error message

The error message as it was confused me when the value was 'enabled' and
there was also a module parameter named 'enabled'.

    enabled is not a valid boolean.  Valid booleans include: yes, on, 1, true, ...

So by clearly describing it as a value, the confusion would have been
avoided.

    The value 'enabled' is not a valid boolean.  Valid booleans include: yes, on, 1, true, ...

bcoca committed:
fixes to virtual facts

fixes #36038

stefanhorning committed:
[cloud][aws] Add metrics_collection options to ec2_asg module (#35180)

* Added metrics_collection management to ec2_asg module to switch ASG metrics on and off.

* Fixed typo in documentation of ec2_asg module

* Removed extra blank line in ec2_asg.py

* Docs fixes for ec2_asg module

* Added integration test for ec2_asg metrics flag

* Trying different syntax for ec2_asg test

gundalow committed:
Fix more docs vs argspec (#36048)

nitzmahone committed:
requested status updates/removals of punted items

mkrizek committed:
add_host: check if name or hostname arg is provided

dagwieers committed:
aci_domain: Add missing integration tests (#36051)

bcoca committed:
corrected dump signature

dagwieers committed:
aci_aep_to_domain: Add missing integration tests (#36071)

dagwieers committed:
aci_domain_to_vlan_pool: Add missing integration tests (#36079)

mattclay committed:
Restore azure_rm_sqlserver test in CI.

dagwieers committed:
ACI docs: Add a section about common error messages (#36088)

Since we pass information directly to ACI, we sometimes get error messages back to the user that require some additional information or context.

This PR includes:
- Changes to the default error output so the error is easier to find
  when searching in e.g. Google
- 3 specific error messages that we have encountered and has confused
  our users before

Commit:
fix nxos_interface_ospf issues (#35988)

* fix nxos_interface_ospf issues

* review comment

Akasurde committed:
VMware: re-enable vmware_guest_powerstate tests (#36092)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
Corrected the helm module documentation to the correct default state (installed -> present). (#35950) (#35951)

gundalow committed:
Fix YAML issue (#36103)

dagwieers committed:
ACI docs: Add more links to resources and community (#36101)

This PR includes:
- Adding more resources to each section
- Improving wordings
- Add links to the Ansible ACI community pages

dminca committed:
provide detailed description of module (#36107)

Fix vague description of module.

Resolves:
Related:
Signed-off-by: Daniel Andrei Minca <mandrei17@gmail.com>

gundalow committed:
Consistant headings (#36112)

dharmabumstead committed:
Core Docs Refactor and Redesign (#36067)

* Docs refactor as outlined in https://github.com/ansible/proposals/issues/79. Moves content into 'guides'; refactors TOC; fixes CSS; design tweaks to layout and CSS; fixes generated plugin, CLI and module docs to fix links accodingly; more.

* Adding extra blank line for shippable

sivel committed:
Prevent reparenting a block with itself (#36075)

* Prevent reparenting a block with itself

* Move __eq__ to Block, to avoid some unexpected problems

Commit:
Upgrade to openshift client 0.4.0 (#35127)

Qalthos committed:
Acquire `display`ed messages from forked process (#36064)

Also restore verbosity value from loaded `play_context`

bcoca committed:
skip aws inventory test if no boto

newswangerd committed:
[WIP] Add ios smoke tests (#35972)

* added basic ios_smoke files

* added smoke tests for network.common.config

* added tests for network.common.utils

* added remainder of test cases

dagwieers committed:
aci_aep_to_domain: Fix query all objects (#36131)

This PR includes:
- A fix to the module where domain_type without domain caused havoc
- A fix to the integration tests when querying all binding objects
- Improvements to integration tests

dharmabumstead committed:
Followup to docs refactor pull request #36067 - fixes gitignore and c… (#36120)

* Followup to docs refactor pull request #36067 - fixes gitignore and cleanup in makefile; removes some generated files; moves a straggler to the appropriate subdirectory.

* Fixed some stragglers

* Removed redundant module entries

* Delete generated RST files.

pfink committed:
Fix typo in vars.rst (#35098)

bcoca committed:
removed gnerated files

@dharmabumstead, probably need a few more of these

bcoca committed:
ensure cli dir exists before saving files to it

dharmabumstead committed:
Fixes broken nested tables in module docs (#36144)

IPvSean committed:
adding support for loopback interface (#36141)

* adding support for loopback interface

currently the loopback interface lo is not supported with vyos_l3_interface, this commit fixes that.  Right now there is a limit of loopback interfaces to just lo, if you want more interfaces you need to use a dummy interface https://wiki.vyos.net/wiki/Dummy_interfaces

* fixing spacing as per pep8 test

fixing issues for sanity test

lib/ansible/modules/network/vyos/vyos_l3_interface.py:120:15: E111 indentation is not a multiple of four
lib/ansible/modules/network/vyos/vyos_l3_interface.py:122:15: E111 indentation is not a multiple of four
lib/ansible/modules/network/vyos/vyos_l3_interface.py:126:19: E111 indentation is not a multiple of four
lib/ansible/modules/network/vyos/vyos_l3_interface.py:128:19: E111 indentation is not a multiple of four
lib/ansible/modules/network/vyos/vyos_l3_interface.py:131:19: E111 indentation is not a multiple of four
lib/ansible/modules/network/vyos/vyos_l3_interface.py:133:19: E111 indentation is not a multiple of four
lib/ansible/modules/network/vyos/vyos_l3_interface.py:137:15: E111 indentation is not a multiple of four
lib/ansible/modules/network/vyos/vyos_l3_interface.py:140:15: E111 indentation is not a multiple of four
lib/ansible/modules/network/vyos/vyos_l3_interface.py:145:15: E111 indentation is not a multiple of four
lib/ansible/modules/network/vyos/vyos_l3_interface.py:148:15: E111 indentation is not a multiple of four

* ugh, missed on spacing issue

* getting rid of continuation lines, the CI system does not like it

LindsayHill committed:
Fixed Arista typo in docstring (#36139)

lachlancooper committed:
Fix description for rabbitmq_policy (#36149)

This was copied from rabbitmq_vhost.

Commit:
Changing Lenovo Inc to Lenovo and update License file to be consistent. (#36155)

winotu committed:
File Module docs, adding mode permission example with sticky bit  (#36151)

* File Module docs, adding mode permission example with sticky bit

dagwieers committed:
ACI: Fixes to domain parameters dependencies (#36158)

We identified an incompleteness to parameter dependencies that affects
querying all domain-related objects.

This PR also includes:
- Improvements to integration tests
- Add missing vm_provider types
- Fix examples

This relates to #36131

samdoran committed:
Add unit tests for edgeos_command (#35861)

gundalow committed:
Ensure docs are valid before checking deprecation (#36160)

dagwieers committed:
aci_vlan_pool: Improve integration tests (#36173)

This PR includes:
- More elaborate integration tests
- Split static and dynamic VLAN pools in separate files

gundalow committed:
Notice in 2.6 remove in Ansible 2.10 (#36191)

Update the list of allowed removed_in for modules

mattclay committed:
Add additional support code in ansible-test.

jborean93 committed:
win_updates: fix regression when using string cat names (#36015)

jborean93 committed:
win_domain_controller: optionally add -ReadOnlyReplica if true (#36017)

jborean93 committed:
win_uri: fix creates/removes option (#36016)

bcoca committed:
Inv export (#36188)

* add export option

* added 'export mode' to ansible-inventory

this optimizes the output for exporting inventory vs representing the 'ansible view'

fixes #30877

* added group priority when needed

bcoca committed:
fix loop_control templating (#36124)

* fix loop_control templating

- properly template ALL properites for loop_control
- loop_control inherits from base, it should not, but needs validate/attribute functionality

fixes #24719

mattclay committed:
Initial OpenShift integration test support. (#36207)

Based on integration tests from chouseknecht for openshift_raw.

dharmabumstead committed:
Relocate ACI guide and add to TOC (#36208)

bcoca committed:
never tags docs and test (#35964)

* fufilling the promise

* moved changes

* Copy edit

dagwieers committed:
ACI: Avoid using 'previous' with query operations (#36181)

When ACI modules are being used for querying MOs, we should not return
the previous state, as there is no previous state, there's only the
current state.

This impacts a lot of tests that were used to testing the current state
as 'previous'.

newswangerd committed:
added support for --testcase flag in ansible-test (#36134)

* added support for --testcase flag in ansible-test

* fixed command format

* added tab completion

* fixed sanity issues

* added documenation for --testcase

* don't autocomplete when multiple modules are selected

bcoca committed:
better sudo/su deprecation on config entries (#36074)

* better sudo/su deprecation on config entries

fixes #32130

* typo fixes

* Copy edit

* Fixed more instances

abadger committed:
* Fix for setting remote_tmp as a subdirectory of a system temp dir (#36143)

Fixes #35666

* Check that system_temps is all absolute paths

fantashley committed:
Update VERSION to match ansible.release (#36212)

dagwieers committed:
aci_encap_pool: Standardize on 'pool_allocation_mode' (#36215)

dharmabumstead committed:
Fixed typo in link. (#36220)

abadger committed:
Final fix for #35666  Previous PR (#36143) was merged prematurely.

abadger committed:
Fix line too long

richlv committed:
Update zabbix.py (#36126)

minor style fixes in module documentation fragment of zabbix

mikewiebe committed:
Fix nxos_linkagg tests (#36175)

pdellaert committed:
Updating URL to Nuage Python VSPK documentation (#36213)

Fixes ansible/ansible#35766

dharmabumstead committed:
Cleaned up some scenario guide titles and TOC placement. (#36225)

gundalow committed:
Add stub porting_guide_2.6.rst (#36190)

* Add stub porting_guide_2.6.rst

* Typo fix

emonty committed:
Add a module_utils OpenStack Cloud constructor (#20974)

Start using this to construct shade OpenStack Cloud objects in a
consistent manner. This will let us centralize things like dealing with
password arguments and whatnot. It also allows us to introduce the
ability to pass a fully formed config dict directly to the module.

Migrate all OpenStack modules to use openstack_cloud_from_module.

Have it return the shade library since it's responsible for
importing shade and shade is needed for the exceptions.

Only pull specific OpenStack arguments for the constructor

Rather than passing **module.params to the shade constructor, pull out
only the values that make sense. This should prevent the issues with
module parameters stepping on shade parameters.

Replace module.params.pop with module.params.get

We don't need to pop these anymore since the shade constructor is now
using opt-in values.

Using real urls is ungood. Use example.com domains. Also, get rid of the
antiquated port numbers.

Qalthos committed:
Alter ASA regexes to be byte-strings (#36176)

pilou- committed:
vars.clean: remove unused method inject_facts (#34838)

* vars.clean: remove unused method inject_facts

Removing this unused method seems better than fixing the typo.

* Fix typo in comments

ryansb committed:
[cloud] Fix inconsistent case of `Allocation_id` in NAT Gateway outputs (#36058)

alikins committed:
Fix ansible-console always asking for vault passwords

ansible-console was calling setup_vault_secrets
twice. Once directly and once via _play_prereqs()

The direct invocation was not setting auto_prompt=False.

However, the direct invocation isn't need at all so
this removes it so only _play_reqs() is used.
That fixes the unrequested vault password
prompting.

Fixes #33027

abadger committed:
Normalize usage of temp and tmp on tmp (#36221)

* Normalize usage of temp and tmp on tmp
* Rename system_tmps system_tmpdirs
* Add ANSIBLE_REMOTE_TMP spelling of environment variables

nwgarner committed:
Update ios_ping.py to allow for count > 70. (#36142)

* Update ios_ping.py to allow for count > 70.

Find 'Success...' output rather than relying on list index.  Index -1 won't pass unit test.

caphrim007 committed:
Reduces the unit test time of select bigip modules (#36256)

The modules in this patch include waits that need to happen to ensure
something is correctly configured on a BIG-IP. These waits were
raised as an issue in a recent ansible-testing meeting.

This patch eliminates the waits by mocking time.sleep

sivel committed:
Add several new doc<->arg_spec checks (#36247)

* Add several new doc<->arg_spec checks. See #18183

* Update ignore.txt for validate-modules

michaelnugent committed:
False matches the docs and intentions of the previous developer (#35762)

deiwin committed:
Fix Porting Guide links for Ansible 2.5 (#36185)

The current links result in 404 Not Found. Also changed protocol to HTTPS.

sivel committed:
Arg spec doc comparison fixes (#36265)

* Ensure we merge doc fragments early, for comparisons

* Perform boolean conversion from arg_spec data too

* Update valdiate-modules ignore.txt due to recent changes

dagwieers committed:
aci_firmware_source: Add integration tests (#36246)

This PR includes:
- Rename 'protocol' parameter to 'url_protocol'
- Add limited integration tests (as we have no public firmware access)
- Add missing examples

dagwieers committed:
aci_taboo_contract: Add integration tests and examples (#36276)

This PR includes:
- Missing integration tests
- Missing examples
- Two typo's in other integration tests

mattclay committed:
Import sanity test for main() in Ansible modules.

ryansb committed:
Fix incorrect docs in `ec2_tag` per #35738 (#36268)

Commit:
fix nxos_bgp_af issues (#36147)

* fix nxos_bgp_af issues

* shippable fix

* review comments

* shippable error fix

Akasurde committed:
VMware: Fix DVPG idempotency issue (#35837)

This fixes, cloning operation where template or existing VM
does not have network or DVPG. Also, adds some strict type checking in
network parameters.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
docker_login: Hide password from response (#34491)

Fixes: #32869

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: New module: vmware_guest_disk_facts (#36026)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ganeshrn committed:
Add eos_command doc with more examples (#36245)

* Add eos_command doc with more examples

*  More examples using cli and eapi transport
*  Example on how to handle json output
*  Example on handling result output in wait_for

* Fix typo

Akasurde committed:
VMware: New module: vmware_guest_snapshot_facts (#36002)

This module gathers facts about VM's snapshots.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mattclay committed:
Truncate some long messages sent to a TTY.

Can be overridden with the --truncate option.

dagwieers committed:
aci_domain_to_encap_pool: Add integration tests (#36217)

Akasurde committed:
Misc. typos in ovirt_host_networks (#36287)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

bcoca committed:
updates to porting guide (#36219)

* updates to porting guide

(cherry picked from commit 7a6d6870d4633ae38a14cc799521b5fc1c0320fd)

* Edits to correct spelling, grammar, and clarify wording.

* Fixed typo

anchat1990 committed:
Add information on the various package states (#35882)

mattclay committed:
Add missing validate-modules ignore entries.

kbreit committed:
Improved language around XML description. (#36263)

* Improved language around XML description.

* Moved details on second line

ThomasADavis committed:
int("255.255.255.0" does not work, fixes issue #33867 (#36271)

newswangerd committed:
re enabled tests that were fixed in  ansible/ansible-modules-core#5008 (#36257)

kbreit committed:
Improved language around in_vpc parameter. (#36137)

bcoca committed:
remove extra fields from debug output

fixes #35493

updated tests

bdowling committed:
Fix None stringification call to send_command (#36254)

send_command already performs the to_bytes safely on prompts (checking
for None).  Without this check the literal 'None' became a subprompt trigger!

Fixes #35662

mkrizek committed:
facts: fix device uuid's on el6 (#36128)

* facts: fix device uuid's on el6

Fixes #36077

mikewiebe committed:
Fix structured output bug (#36193)

bcoca committed:
updates to module testing (#36043)

* updates to module testing

gives those using internal modules an alternative

* Copy edit

gundalow committed:
Depreciate aos (#36029)

* Deprecate Apstra's aos_* modules

These modules don't work with AOS 2.1 or higher.
They will be replaced non-upstreamed modules

* Correct list

gundalow committed:
Update ignore to deal with renamed files (#36323)

mikewiebe committed:
Fix nxos_system tests (#36201)

* Fix nxos_system tests

* Add debug connection plugin info

* Move sanity test under common

rahushen committed:
Fixes for the N3500 platform that uses the A8 image (#36261)

* fix nxos_l3_interface tests as n35 doesn't support ipv6

* add terminal dont-ask to nxos_feature and nxos_lldp

* put interfaces in L2 mode for N35

* fix nxos_feature unit-tests

mikewiebe committed:
Remove kickstart_image_required check (#36319)

webknjaz committed:
Drop duplicate imports from apt_repository module (#36320)

* Drop duplicate imports from apt_repository module
* Drop json import from apt_repository module

Akasurde committed:
Refactor DigitalOcean modules (#35934)

- Updated documentation fragment
- Updated Argument spec
- Use common REST class for API calls

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Fix digital ocean issues (#36347)

Remove module from ignore lists and some documentation fix in
digital_ocean_block_storage and digital_ocean module.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

BondAnthony committed:
Fix bug in DigitalOcean module_util (#36352)

mattclay committed:
Support ansible-test --truncate with delegation.

trishnaguha committed:
nxos_system test fix inventory_hostname_short (#36359)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

trishnaguha committed:
fix nxos_config json pipeline error (#36236)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

trishnaguha committed:
Set hostname to 'switch' on nxos_system (#36373)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

fgeek committed:
hashi_vault: fix typo (#36368)

Akasurde committed:
VMware: Refactor vmware_guest_tools_wait (#36098)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: Make a common API for hostsystem params (#36314)

This fix adds a common API for getting host system managed object
from either cluster name or host system.

Fixes: #36010

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
eos_vlan: Modified logic to allow for more than 6 interfaces to be assigned to … (#35568)

* Modified logic to allow for more than 6 interfaces to be assigned to a vlan

* arista output to json.  removing logic for string size.

felixfontein committed:
Letsencrypt authz deactivation (#36362)

* Allow to deactivate authz objects. Currently only after success.

* Making sure cleanup is done even when module fails (except if fetch_url() fails).

* Make deactivate_authzs eat exceptions so that all authzs are deactivated in case of errors.

dagwieers committed:
aci_rest: Fix error handling and improve documentation (#36295)

This PR includes:
- A fix for a recently introduced issue wrt. error handling
- Added integration tests for provoked errors
- Influence standard return values using aci library for aci_rest
- Add proxy support documentation
- Documentation update related to #34175

Commit:
Use pycrypto backend, rather than python-cryptography, on Ubuntu Precise

dagwieers committed:
aci_rest: Remove test data from validation test (#36381)

Oops

ganeshrn committed:
Update network debug troubleshooting guide (#36288)

* Update network debug troubleshooting guide

Fix #35914

Command timeout and connection timeout error messages
are dsiplayed in log file instead on console.
Update the same in troubleshooting guide.

* Update example error

* Fix CI issues

* Fix more CI failures

* More fixes

* Fix review comments

* Fix more review comments

* Copy edit

richlv committed:
Update zabbix_host.py (#36127)

fix tls documentation; minor style fixes

abadger committed:
Fix the script and patch plugins tempfile ownership

Unified tmp accidentally removed the containing tmpdir from the list of
files to fix the permissions on when we're becoming a different
unprivileged user.  This resulted in a visible bug for script but not
for patch.  This is because patch also uploads the module to the same
temporary directory and the uploaded module also ends up calling
fixup_perms2() which includes the temporary directory.  So by the time
patch needs to access the temporary patch file, the directory is
appropriately set.

script's breakage was visible because script does not upload a module
(it's akin to raw in this way).  Therefore, we only call fixup_perms2()
once in script and so leaving out the tmpdir in script means that the
containing directory never has its permissions set appropriately.

Fixing both because it does not cause an extra round trip for patch so
any speedup would be minimal and it's better to fix the perms as close
as possible to where we know we need it.  Otherwise, changes to
seemingly unrelated code later could end up breaking it.

Fixes #36398

dagwieers committed:
mail: Fix cannot concatenate 'str' and 'int' objects

Fix an error in callback plugin when exceptions are being returned on failure.

mattclay committed:
Add option to hide sensitive ansible-test output.

This option is enabled automatically on Shippable.

s-hertel committed:
[cloud] If inventory file isn't able to be parsed by aws_ec2, raise an AnsibleParserError (#36387)

* If inventory file isn't able to be parsed by aws_ec2, return an empty dict instead of None

* Raise an AnsibleParserError instead

* remove extra lines

* aws_ec2 inventory plugin - fix path matching logic

mattclay committed:
Fix ansible-test --redact option with delegation.

dagwieers committed:
aci_aep: Add missing integration tests (#36414)

This PR includes:
- Add missing aci_aep integration tests
- Fix small aci_rest issue

dagwieers committed:
ACI: Add missing examples to module docs (#36417)

This PR includes:
- A cleanup of documentation examples for ACI modules
- A small fix in aci_rest integration test

dagwieers committed:
aci_encap_pool: Fix vxlan integration tests (#36418)

Missing cleanup would cause subsequent runs to fail when interrupted.

Commit:
fix nxos_bgp_neighbor issues (#36318)

webknjaz committed:
Fix portage mod usage of jobs and load-average opt

In short, it enables portage module to emit emerge
command with --jobs and --load-average options
with and without argument.

To emit emerge with such CLI argument without
a value, the user must set the corresponding
module option to 0.

By default, if these arguments are missing from
playbook they are omitted.

Misc changes:

* Use to_native to ensure crosspython compat

* Adjust jobs and LA options to accept 0 as reset

* Add docstring to emerge_packages

* Explicitly note in doc that False will work as 0

PR  #36355 by @webknjaz

jctanner committed:
Do not cache the loop item label so that it will update with each item

jtyr committed:
Fix default value of gpgcheck in doc (PR #36267)

gdpak committed:
vyos_config IndexError in sanitize_config (#36375)

* Bug in del(list) logic. Say you have a list of 4 elements a[0-3]
and you have to remove index 1 and 3, if you remove index 1 first
then list is cut short a[0-2] and does not have index 3

Fix: Remove indexes in reverse sorted order e.g. 3 and 1 in above
example so that order of indexes remain preserved even after deleting

fix is to remove indexes in reverse sorted order

* Add test cases for failed case

bcoca committed:
now get_url defaults to module temp dir (#36218)

* now get_url and other modules default to module temp dir

also fixed 'bare' exception

* allow modules to work with older versions

* updated docs per feedback

bcoca committed:
hide uneeded fields for callbacks (#36259)

* hide uneeded fields for callbacks

fix selective instead of pushing uneeded fields to the methods

* piipii

ryansb committed:
Add AWSRetry when describing VPCs to help stabilize integration tests (#36264)

* Add AWSRetry when describing VPCs to help stabilize integration tests

* Add retry on create_tags because it is possible to reach this API call before the VPC is finished creating

brunocalogero committed:
added aci_fabric_node integration test (#36422)

* added aci_fabric_node integration test

* Update main.yml

bcoca committed:
catch all cases of no inventory when asked to fail

fixes #36034

ryansb committed:
Change to longer interval jittered backoffs for efs facts (#36266)

* Change to longer interval jittered backoffs for security group fact gathering

* Add the ECS throttling error code to the list of things to retry on

trishnaguha committed:
fix vyos_l3_interface adding multiple addresses to interface (#36377)

* fix vyos_l3_interface adding multiple addresses to interface

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* add test

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

mattclay committed:
Fix path handling in validate-modules sanity test.

brunocalogero committed:
added aci_vpc_protection_group integration test (#36448)

added aci_switch_policy_vpc_protection_group integration test

mattl committed:
Update vault.rst (#36328)

hunter2 not hunter42. http://bash.org/?244321

Klaas- committed:
[Doc] Update Conditional Imports (#35492)

* [Doc] Update Conditional Imports

facter/ohai is not needed for the ansible internal facts
also fix that os_family for centos is redhat

* Copy edit.

mattclay committed:
Add support for enhanced code-smell tests. (#36332)

* Add support for enhanced code-smell tests:

- Path selection handled by ansible-test.
- Optional path filtering based on extension.
- Optional path filtering based on prefixes.
- Optional lint friendly output.

* Enhance no-assert code-smell test.
* Enhance no-tests-as-filters code-smell test.

dagwieers committed:
aci_spvpg: Various fixes to integration tests/modules (#36458)

* aci_spvpg: Various fixes to integration tests/modules

This PR includes:
- A fix in module_utils aci.py
- Various fixes in integration tests

* Fix typo

resmo committed:
cs_user: fix return user_api_secret for ACS v4.10 and later (#36447)

dagwieers committed:
aci_iplpg: Various fixes to integration tests/module (#36460)

samdoran committed:
Add force option to systemd module (#35925)

* Add force option to systemd module

* Use multi-line YAML syntax in examples

* Add version_added for new option

mikewiebe committed:
Fix nxos_banner module for unstructured output (#36411)

* Fix nxos_banner module for unstructured output

* Refactor and bug fixes

* Fix pep8 error

Commit:
fix nxos_bgp_neighbor_af issues (#36472)

brunocalogero committed:
aci_static_binding_to_epg fixes  (#36478)

* aci_static_binding_to_epg fixes (documentation and logic for interface_mode)

* additional fixes after review

mattclay committed:
Fix PEP 8 issue.

Akasurde committed:
VMware: Add missing state param in argument spec (#36339)

This fix adds missing argument parameter called 'state'
and its documentation.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

LindsayHill committed:
fixed conditional error message (#36476)

goshkis committed:
vApp properties support (#32579)

* Adding support for vApp properties.
* vm specification updated only if changes have to be applied. I.e. subsequent updates with the same data will not trigger changed state
* Auxiliary variables renamed, hope this makes the code more readable
* Integration tests changed - re-adding the same properties test not implemented, but tested on real vCenter deployment
* fixing documentation "version_added" for the feature
* Addressing reviewers comments #2:
* documentation updated with the only meaningful value for "option" attribute - "remove"
* Fixed improperly handled case when user requested "add" operation for existent property
* vApp configuration is updated only with properties that contains changes, not with all properties requested by user

trishnaguha committed:
fail module when config is invalid and jsonrpc doesn't return error (#36482)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

jborean93 committed:
fetch: set fail_on_missing: True as default as per docs (#36469)

* fetch: set fail_on_missing: True as default as per docs

* Updated docs for fetch to say behaviour was changed in 2.5 and updated tests

doubleplush committed:
Wait for VM state to reach poweredoff when state: shutdownguest (#31669)

This change adds the optional wait_for_state_change argument to the
vmware_guest, vmware_guest_powerstate module, which allows for module
completion to be blocked when using the shutdownguest state until the
VM has reached the poweredoff state.

Fixes: #28498

Signed-off-by: Jim Gu <heming.gu@mercurygate.com>
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mikedlr committed:
aws ssm parameter lookup - change to return Nones for missing variables (#36456)

* aws ssm parameter lookup - change to return Nones for missing variables

* aws ssm parameter lookup - fix error case message to dump response

* aws ssm parameter lookup - fix integration test cases

trishnaguha committed:
fix nxos_pim module doc (#36505)

* fix nxos_pim module doc

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* address review

s-hertel committed:
[cloud] Retry WAF actions on WAFStaleDataException (#36405)

Add a util to run functions with AWSRetry to retry on WAFStaleDataExceptions and update ChangeToken for each attempt

Commit:
Fixes #12353 #20977 adds new option to galaxy cli to preserve scm meta (#34642)

trishnaguha committed:
nxos_vrf fix (#36494)

* nxos_vrf fix

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* Address review

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

s-hertel committed:
[ec2_vpc_net_facts] Fix UnsupportedOperation for regions other than u… (#35302)

* [ec2_vpc_net_facts] Fix UnsupportedOperation for regions other than us-east-1

* Make fix more Pythonic

* Fix the exception handling for the module

samdoran committed:
Add note to use script module rather than inline here documents (#32863)

* Add example on how to use a here document with shell module

* Remove here doc example and add note to use script module instead

orthanc committed:
Better handling of absent AWS SES identity notification information. (#36354)

* Better handling of absent AWS SES identity notification information.

Fixes #36065

aws_ses_identity module now handles the cases where information about
the notification setup for the identity isn't returned by the AWS api.

This seems to happen in an edge case, believed to be eventual
consistency on registering new identities. So this case is treated
as if has been no notification setup for the identity yet.

Also fix 2 flake8 warnings in the module, a missing newline and unused
import.

* Increase the Boto Retries on SES APIs to deal with throttling.

This should address the unstable integration test failing due to
parallel runs in shippable hitting AWS throttling.

* Add retries loading SES details for inclusion in successful response.

There seems to be an eventual consistency behaviour with identity
registration. It's possible to still get no identity back after
registration.

This can cause failures in the shippable builds. This should fix that by
creating a retry of retrieving the identity information after
registration.

A similar retry loop has been added to notification attributes to ensure
this doesn't suffer from the same failure.

* Add missing sleep in get_notification_attributes to avoid busy loop.

mkrizek committed:
Add integration test for #36430 (#36432)

gundalow committed:
Fixup VMware module docs issues (#36409)

Fix various argspec vs documentation issue

bcoca committed:
remove query/q (#36315)

* remove query/q

remove and fixed tests for removed func

* paren fix

abadger committed:
Notify cyberark-bizdev for cyberark plugin and module tickets.

tgates81 committed:
find.py module: Added depth: option to specify how many level deep to… (#36205)

* find.py module: Added depth: option to specify how many level deep to traverse directories.

* find.py module: depth: added correct version_added value.

* find.py module: depth: Default value is None.

sivel committed:
Re-use logic from StrategyBase._load_included_file in StrategyModule.run for free and linear (#36470)

This improves include_role performance and recursion limits

abadger committed:
Revert "remove query/q (#36315)"

This reverts commit b47d2e07e15a8a15307f146e0c12e70e5e4ca0cd.

query is a feature for 2.5.  We're not reverting it now.

dagwieers committed:
ACI: Various changes to module documentation and guide (#36516)

* ACI: Various changes to module documentation and guide

This PR includes:
- We moved the object class information to the notes
- Add version information to guide chapters
- Add generic note to modules with reference to ACI guide
- Reference known issues in aci_rest documentation
- Remove module_utils function docs from modules
- Indicate which parameters are not required for querying all objects
- Added missing RETURN information

* Fix copyright strings

* Remove aci_domain_to_encap_pool.py for v2.5

* More updates

* PEP8 fix

* Improve listings of parameters/return values

dagwieers committed:
aci_domain_to_encap_pool: Remove integration tests (#36535)

julienvey committed:
[cloud] ec2_group: fix check mode when using tags (#36503)

Fixes #36412

brunocalogero committed:
minor fix to exit_json for aci_static_binding_to_epg.py (#36540)

* minor fix to exit_json for aci_static_binding_to_epg.py

* The aci.exit_json() call doesn't need arguments

brunocalogero committed:
add aci_static_binding_to_epg integration tests (#36542)

s-hertel committed:
[cloud] catch NoCredentialsError in boto3_conn (#36537)

brunocalogero committed:
changed cisco mail to normail hotmail (#36546)

samdoran committed:
Check for regexp match when using insertbefore or insertafter (#36474)

Add tests to cover this scenario
Fixes #36156

dagwieers committed:
aci_interface_policy_leaf_policy_group: Fix test (#36551)

dagwieers committed:
Add missing copyright notice

I did a large part of the unzip implementation and gtar functionality.

julienvey committed:
efs: improve documentation about the update of an efs (#36473)

jborean93 committed:
win_feature: better error handling to make it easier to debug issues (#36491)

* win_feature: better error handling to make it easier to debug issues

* removed ignroed pslint rules that are no longer needed

sivel committed:
Fix types when evaluating interpreter (#36541)

* Fix types when evaluating interpreter. Fixes #36536
* Rename variables that contain bytes to b_*
* Get rid of to_text() and to_bytes() calls that do nothing (because the
  data is already the proper type)

mattclay committed:
Upgrade more code-smell tests. (#36560)

* Enhance no-dict-* code-smell tests.
* Enhance no-basestring code-smell test.
* Enhance no-get-exception code-smell test.
* Enhance empty-init code-smell test.
* Enhance required-and-default-attribute test.
* Remove unused code-smell test.

gratuxri committed:
Updated vmware URL for guest_ids (#36511)

richlv committed:
Update zabbix_maintenance.py (#36123)

clarify that removal happens by name - while expected, it's good to be explicit

mattclay committed:
Update more code-smell tests. (#36570)

* Enhance use-compat-six code-smell test.
* Enhance use-argspec-type-path code-smell test.
* Enhance replace-urlopen code-smell test.
* Enhance boilerplate code-smell test.
* Enhance no-underscore-variable code-smell test.

ryanpetrello committed:
tower cred: implement credential /api/v1/ kind compatability

ryanpetrello committed:
tower cred: filter user name lookup by the proper key

bcoca committed:
protect against plugins using verify incorrectly

assume false on any errors

s-hertel committed:
aws_ec2 inv plugin: remove extra call to verify_file (#36588)

* Remove extra call to verify_file

* Allow next inventory plugin to be tried if the file isn't verified

ryanpetrello committed:
tower cred: update kind options in documentation

sdodsley committed:
Initial commit for Pure Storage FlashBlade module (#32467)

Commit:
Network Getting started docs (#36337)

* first draft of getting started guide

Commit:
Bump OpenShift client req to 0.4.3. Fixes openshift_scale. (#36601)

sivel committed:
Start of tests for modify_module, specifically to ensure proper shebang replacement on old style modules (#36602)

julienvey committed:
ec2_group:documentation explicit a behaviour for sg rules (#36586)

brunocalogero committed:
added additional examples for aci modules (#36612)

* added additional examples for aci modules

* small fix on aci_fabric_node

* removing jina2 for clarity

brunocalogero committed:
aci example minor fix testing actual real life scenario examples (#36613)

mkrizek committed:
yum: fix yum version comparison (#36573)

orthanc committed:
Fix documentation of validate_certificates option in aws doc_fragment.

Unignore sanity test failures for AWS modules caused by this common
fragment.

dminca committed:
fix erroneous example in inventory (#36625)

After initializing a list in both Python2/Python3 with below elements,
I've tested the indexing and realised that the examples provided in the
documentation are erroneous.

As a result, update the examples with correct values in order to avoid
any future confusion.

Resolves:
Related:
Signed-off-by: Daniel Andrei Minca <mandrei17@gmail.com>

pilou- committed:
filesystem: fix documentation (#36556)

Gittins committed:
Updated examples aci_contract_subject_to_filter.py (#36626)

Changed examples to reference the correct module name. Previously the examples used "aci_subject_filter_binding".

Commit:
Adds k8s_raw, openshift_raw tests (#36228)

trishnaguha commented:
@tomilukkarinen Thanks for your PR. Looks likes you are adding feature pull request. We accept only bugfixes for `stable-2.4` branch. Please reopen the PR against `devel` branch. Also you'll need to fix your branch since you have commits that is not related to your PR.

sivel committed:
Add instructions for creating backport PRs (#36593)

* Add instructions for creating backport PRs

* Update development_process.rst

simpler workflow

* Update development_process.rst

added origin note

* A few adjustments to clarity, use backport instead of cherry pick in branch name

* Address formatting issue

* fetch isn't a flag

* Fix rst formatting

bcoca committed:
debug should have item (#36534)

* debug should have item

* item should only be present when var

gundalow committed:
Notify acozine & gundalow on Network Docs (#36655)

Now we have a docs directory notify and add labels on any network docs
PRs

dagwieers committed:
Add anchors to some guides and all module categories (#36642)

* Add anchors to some guides and all module categories

This is required if we want to use *absolute* :ref: references instead of *relative* :doc: references.

* Update the Cisco ACI Guide reference

* Add `aci_guide` anchor

* Add `network_guide` anchor

* Add category anchor

* Improve readability

* Fix small typo

Commit:
Fixed indentation error in `facts-demo.yml` example. (#36645)

* Fixed indentation error in `facts-demo.yml` example.

`content` and `dest` should be indented under the `copy` module.

* Updated indentation of copy module in `facts-demo.yml` example.

Updated indentation of copy module in `facts-demo.yml` to be inline with its name.

dagwieers committed:
ACI: Improve module index short_description (#36665)

dagwieers committed:
guide_aci: Fix a few references (#36666)

dagwieers committed:
docsite: Add 'Edit on GitHub' for module docs (#36667)

This is something I always wanted, a 'Edit on GitHub' button for module
documentation.

I also removed the additional statement in the footer with instructions
on how to edit the module documentation.

PS The links go directly into the GitHub file editor now !

trishnaguha commented:
Closing this PR.
